qiime/core/util.py

```diff
@@ -20,3 +22,36 @@ def decorator(func):
                                  % (func, cls.__name__))
         return func
     return decorator
+
+
+def parse_type(string):
+    # Avoid circular imports
+    import qiime.sdk
+    from .type import Visualization, Properties
+
+    if len(string.split('\n')) != 1:
```

Comment:
Come to think of it, we're disallowing even a single `\n`. Is that desirable? If so, we should probably update the error message and just check `'\n' in string` which'll be faster.